### PR TITLE
Add isApproved to v2 expenses queries

### PR DESCRIPTION
**Pull request by sbinlondon**
To fix https://sentry.io/organizations/open-collective/issues/1983864155/?project=1736806

### CODE REVIEW

**Review by znarf — COMMENTED** [review]
Why would collective be null? How can we prevent this?

**Comment by sbinlondon** [comment]
Perhaps while the page is loading? I checked all the parent components and we're definitely passing down `collective` where it is needed in all of these components (`ExpenseSummary`, `ExpensePayeeDetails`, etc...)

**Review by Betree — COMMENTED** [review]
All these components indeed have a `isLoading` flag that specify that the data may be missing if it's loading. But in `ExpensePayeeDetails` I agree that the check is a bit strange because we check if the host exists above (and if the host is set, you'd normally expect the collective to be there too).

But if we can confirm that's just because of a loading state, I'm all good with this.

**Comment by znarf** [comment]
If it's because of a loading state, I prefer that we're not rendering (and render a loading state instead). I'm not comfortable with the idea of having a required object optional.

**Comment by sbinlondon** [comment]
> If it's because of a loading state, I prefer that we're not rendering (and render a loading state instead). I'm not comfortable with the idea of having a required object optional.

Should we merge this as a quick fix and arrange the loading state in a follow up PR? It will probably require a bit more rejigging of the components.

**Comment by sbinlondon** [comment]
The issue was that `isApproved` was not available in the v2 queries. It's been added so can remove optional chaining and not worry about the loading state.

**Comment by znarf** [comment]
> The issue was that isApproved was not available in the v2 queries. It's been added so can remove optional chaining and not worry about the loading state.

Unless I'm missing something `Cannot read property 'isApproved' of undefined` means that `collective` is null not that `isApproved` is null. So, while the latest changes are great, they're not fixing the initial issue.

**Review by znarf — APPROVED** [review]
I'm good with this PR now but I don't think it's fixing the original Sentry issue.

**Comment by sbinlondon** [comment]
> > The issue was that isApproved was not available in the v2 queries. It's been added so can remove optional chaining and not worry about the loading state.
> 
> Unless I'm missing something `Cannot read property 'isApproved' of undefined` means that `collective` is null not that `isApproved` is null. So, while the latest changes are great, they're not fixing the initial issue.

Right, sorry, I've added that back in case `collective` is null so we won't falsely render 'pending' if anything is missing.

File: components/expenses/ExpensePayeeDetails.js [hunk 1]
```diff
@@ -232,8 +232,8 @@ ExpensePayeeDetails.propTypes = {
   /** Disable border and paiding in styled card, usefull for modals */
   borderless: PropTypes.bool,
   collective: PropTypes.shape({
-    isApproved: PropTypes.bool,
-  }),
+    isApproved: PropTypes.bool.isRequired,
+  }).isRequired,
```
Comment (znarf) [inline components/expenses/ExpensePayeeDetails.js:236]: I think it's false to say `collective` is required because it's not if `isLoading=true`

**Pull request merged**